src/test/java/org/folio/consortia/service/ConsortiaConfigurationServiceTest.java

```diff
@@ -0,0 +1,111 @@
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.folio.consortia.utils.EntityUtils.createConsortiaConfiguration;
+import static org.folio.consortia.utils.EntityUtils.createConsortiaConfigurationEntity;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@SpringBootTest
+@EnableAutoConfiguration(exclude = BatchAutoConfiguration.class)
+class ConsortiaConfigurationServiceTest {
+  private static final String CENTRAL_TENANT_ID = "diku";
+  private static final String TENANT_ID = "testtenant1";
+
+  @InjectMocks
+  ConsortiaConfigurationServiceImpl configurationService;
+  @Mock
+  ConsortiaConfigurationRepository configurationRepository;
+  @Mock
+  FolioExecutionContext folioExecutionContext;
+  @Mock
+  ConversionService conversionService;
+
+  @Test
+  void shouldGetConfigValueByGetCentralTenantId() {
+    List<ConsortiaConfigurationEntity> configurationEntityList = List.of(createConsortiaConfigurationEntity(CENTRAL_TENANT_ID));
+
+    when(configurationRepository.findAll()).thenReturn(configurationEntityList);
+    String actualCentralTenantId = configurationService.getCentralTenantId(TENANT_ID);
+
+    Assertions.assertEquals(CENTRAL_TENANT_ID, actualCentralTenantId);
+  }
+
+  @Test
+  void shouldGetConfigValueByGetConsortiaConfiguration() {
+    ConsortiaConfigurationEntity configuration = createConsortiaConfigurationEntity(CENTRAL_TENANT_ID);
+    List<ConsortiaConfigurationEntity> configurationEntityList = List.of(configuration);
+
+    when(configurationRepository.findAll()).thenReturn(configurationEntityList);
+    when(conversionService.convert(configuration, ConsortiaConfiguration.class))
+      .thenReturn(createConsortiaConfiguration(CENTRAL_TENANT_ID));
+    when(folioExecutionContext.getTenantId()).thenReturn("testtenant1");
+    when(folioExecutionContext.getInstance()).thenReturn(folioExecutionContext);
+    Map<String, Collection<String>> okapiHeaders = new HashMap<>();
+    okapiHeaders.put(XOkapiHeaders.TENANT, List.of("testtenant1"));
+    when(folioExecutionContext.getOkapiHeaders()).thenReturn(okapiHeaders);
+
+
+    var expected = configurationService.getConsortiaConfiguration();
+
+    Assertions.assertEquals(CENTRAL_TENANT_ID, expected.getCentralTenantId());
+  }
+
+  @Test
+  void shouldSaveConfigValue() {
+    ConsortiaConfigurationEntity configuration = createConsortiaConfigurationEntity(CENTRAL_TENANT_ID);
+
+    when(configurationRepository.save(any())).thenReturn(configuration);
+    when(configurationRepository.count()).thenReturn(0L);
+
+    configurationService.createConfiguration(CENTRAL_TENANT_ID);
+
+    verify(configurationRepository, times(1)).save(any());
+  }
+
+  @Test
+  void shouldThrowCentralTenantNotFoundErrorWhileSavingConfigValue() {
```

Comment:
should be renamed, ResourceAlreadyExistException is checked here

Reply:
Fixed